Set up domain for STUN server

Follow up to #23

Set up the STUN server to run on a domain that can be configured to round-robin to multiple servers, or at least switched if we find that the current one is down.

~Note that the current setup is on a floating IP so we should be able to reassign to a new server if we need to spin one up, in a pinch~ *Hm, nope: using a floating IP borks the STUN request with:*

```
[ERROR] Other (cause; assertion failed: `left == right`; assertion failed: `(left == right)` (left: `1581330`, right: `554869826`): Unexpected MAGIC_COOKIE: actual=0x00182112, expected=0x2112a442)
HISTORY:
  [0] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:266
  [1] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:295 -- message_type=Type { class: Request, method: Method(650) }, message_len=1, transaction_id=TransactionId(0xA442F3376B9F6D7FC245542D)
  [2] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.14/src/combinator.rs:1368
  [3] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.13/src/message.rs:430
  [4] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.14/src/decode.rs:448
  [5] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/udp.rs:199 -- peer=50.0.92.2:9000
  [6] at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.3/src/error.rs:19 -- original_error_kind=InvalidInput
  [7] at src/transport/udp.rs:358
  [8] at src/transport/udp.rs:193
  [9] at src/channel.rs:176
  [10] at src/error.rs:61 -- original_error_kind=CodecError
  [11] at src/server.rs:340
```

--- a/trin-core/src/socket.rs
+++ b/trin-core/src/socket.rs
@@ -7,7 +7,9 @@ use interfaces::{self, Interface};
 #[cfg(windows)]
 use ipconfig;
 
-const STUN_SERVER: &str = "143.198.142.185:3478";
+// This stun server is part of the testnet infrastructure.
+// If you are unable to connect, please create an issue.
+const STUN_SERVER: &str = "159.223.0.83:3478";
 
 /// Ping a STUN server on the public network. This does two things:
 /// - Creates an externally-addressable UDP port, if you are behind a NAT

